common/src/main/java/com/microsoft/identity/common/internal/dto/AccessTokenRecord.java

```diff
@@ -344,4 +344,23 @@ private boolean isExpired(final String expires) {
     public boolean isExpired() {
         return isExpired(getExpiresOn());
     }
+
+    /**
+     * If the AT has a refresh_on timestamp (typically used for LLT's),
+     * this function will return true after the server recommended refresh
+     * interval has elapsed, prompting the library to kick off a background refresh operation.
+     *
+     * If the AT does NOT have a refresh_on timestamp, this function will always
+     * return false. Fallback to standard token expiration/refresh logic.
+     *
+     * @return Should the library kick off a background token refresh operation for this token
+     * after returning the token to the caller.
+     */
+    public boolean shouldRefresh() {
+        final String refreshOn = getRefreshOn();
+        if(refreshOn != null && !refreshOn.isEmpty()) {
+            return isExpired(refreshOn);
+        }
+        return false;
```

Comment:
It seems like it might potentially be safer (from a calling/understanding POV) to call `return isExpired()` instead of `return false` - thoughts?

Reply:
The way I'm writing the logic in C++ works like this:

1. Get AT from cache
2. Check if AT is expired (if yes, break here and go straight to a synchronous RT renewal to get a valid AT for the app)
3. Check if AT should be refreshed (if yes, create an async RT renewal background request, and add it to our BG task queue)
4. Regardless of step 3, return the current AT to the app

Reply:
In either case, I have no horse in this race! Feel free to commit the suggestion I left in a separate comment if you think that's the better approach!

Reply:
Just seeing this now... let me ponder this a bit and I will plan to finalize your CR tomorrow. Thank you @logdog16!